libfwupd/fwupd-client-sync.c

```diff
@@ -1357,6 +1408,7 @@ fwupd_client_refresh_remote (FwupdClient *self,
 					   fwupd_client_refresh_remote_cb,
 					   helper);
 	g_main_loop_run (helper->loop);
+	g_main_context_pop_thread_default (helper->context);
```

Comment:
this is missing the matching `g_main_context_push_thread_default (helper->context);`

Reply:
Eagle eyes! Fixed, thanks.

Reply:
I wish I could blame my eyes, but I tried it and saw it broken :)